server/server.go

```diff
@@ -488,8 +494,13 @@ func (w *ecdsaWorker) Do(job interface{}) interface{} {
 	}
 
 	keyLoadBegin := time.Now()
-	if key, ok = w.s.keys.Get(&pkt.Operation); !ok {
-		log.Error(protocol.ErrKeyNotFound)
+	key, err := w.s.keys.Get(&pkt.Operation)
+	if err != nil {
+		log.Errorf("failed to load key: %v", err)
+		w.s.stats.logInternalError(pkt.Opcode)
```

Comment:
Interesting thought: If a request fails with `logInternalError` or `logInvalid`, then we don't collect metrics on how long the request took to serve. I don't think we do, at least.

Reply:
That's correct, and that was the intention of the original author: https://github.com/cloudflare/gokeyless/blob/05339602cb41b7859abfd099b9b4dfa4ad19ed9d/server/server.go#L678.

Personally I think it would be valuable to have those timing metrics regardless, with a label for success/failure so we can still slice later if desired. I can make that change now if you agree it would be valuable.

Reply:
I agree.